deploy-agent/deployd/common/stats.py

```diff
@@ -28,69 +49,287 @@ def __exit__(self, stats, sample_rate=1.0, tags=None):
         pass
 
 
-def _add_default_tags(tags=None):
-    """ add default tags to stats """
-    if not __version__:
-        # defensive case, should not be hit
-        return tags
-    if __version__ and not tags:
-        tags = dict()
-    tags['deploy_agent_version'] = __version__
-    return tags
+def create_stats_timer(name, sample_rate=1.0, tags=None):
+    if IS_PINTEREST:
+        from pinstatsd.statsd import statsd_context_timer
 
+        class TimingOnlyStatClient:
```

Comment:
why define a class inside a function?

Reply:
In this case it was done for readability.  In retrospect, that's bad form.  Moved it out.  Thanks!